### PR TITLE
github repo -> helm repo

### PR DESCRIPTION
**Pull request by kalantar**
Addresses: https://github.com/iter8-tools/iter8/issues/1352

Signed-off-by: Michael Kalantar <kalantar@us.ibm.com>

### CODE REVIEW

**Comment by kalantar** [comment]
This PR depends on https://github.com/iter8-tools/hub/pull/31.
It represents a breaking change.
Once the above PR is merged, `testdata/charts/iter8` should be updated using:

```shell
helm pull --repo https://iter8-tools.github.io/hub iter8 --untar
```

File: cmd/launch.go [hunk 1]
```diff
@@ -74,8 +71,35 @@ func addDryRunFlag(cmd *cobra.Command, dryRunPtr *bool) {
...
+func addLocalChartFlag(cmd *cobra.Command, localChartPtr *bool) {
+	cmd.Flags().BoolVar(localChartPtr, "localChart", false, "use local charts")
+	cmd.Flags().Lookup("localChart").NoOptDefVal = "true"
+}
+
+// addChartPathOptionsFlags adds flags related to Helm chart repository
+// copied from
+// https://github.com/helm/helm/blob/ce66412a723e4d89555dc67217607c6579ffcb21/cmd/helm/flags.go
+func addChartPathOptionsFlags(cmd *cobra.Command, c *action.ChartPathOptions) {
+	cmd.Flags().StringVar(&c.Version, "version", "", "specify a version constraint for the chart version to use. This constraint can be a specific tag (e.g. 1.1.1) or it may reference a valid range (e.g. ^2.0.0). If this is not specified, the latest version is used")
```
Comment (sriumcp) [inline cmd/launch.go]: User `majorMinor` version to derive version automatically (e.g., `0.12.x`). Comment out other options.

File: driver/kubedriver.go [hunk 2]
```diff
@@ -300,14 +271,32 @@ func writeManifest(rel *release.Release) error {
...
+	client.ChartPathOptions.CertFile = chartPathOptions.CertFile
+	client.ChartPathOptions.KeyFile = chartPathOptions.KeyFile
+	client.ChartPathOptions.InsecureSkipTLSverify = chartPathOptions.InsecureSkipTLSverify
+	client.ChartPathOptions.Keyring = chartPathOptions.Keyring
+	client.ChartPathOptions.Password = chartPathOptions.Password
+	client.ChartPathOptions.PassCredentialsAll = chartPathOptions.PassCredentialsAll
+	client.ChartPathOptions.RepoURL = chartPathOptions.RepoURL
+	client.ChartPathOptions.Username = chartPathOptions.Username
+	client.ChartPathOptions.Verify = chartPathOptions.Verify
+	client.ChartPathOptions.Version = chartPathOptions.Version
```
Comment (sriumcp) [inline driver/kubedriver.go:290]: Version is the only property that need to be derived.

File: driver/kubedriver.go [hunk 3]
```diff
@@ -300,14 +271,32 @@ func writeManifest(rel *release.Release) error {
...
 
-	ch, vals, err := kd.getChartAndVals(chartDir, valueOpts)
+	client.ChartPathOptions.CaFile = chartPathOptions.CaFile
+	client.ChartPathOptions.CertFile = chartPathOptions.CertFile
+	client.ChartPathOptions.KeyFile = chartPathOptions.KeyFile
+	client.ChartPathOptions.InsecureSkipTLSverify = chartPathOptions.InsecureSkipTLSverify
+	client.ChartPathOptions.Keyring = chartPathOptions.Keyring
+	client.ChartPathOptions.Password = chartPathOptions.Password
+	client.ChartPathOptions.PassCredentialsAll = chartPathOptions.PassCredentialsAll
+	client.ChartPathOptions.RepoURL = chartPathOptions.RepoURL
```
Comment (sriumcp) [inline driver/kubedriver.go:287]: RepoURL is also derived automatically (fixed for now).

File: driver/kubedriver.go [hunk 4]
```diff
@@ -323,46 +312,53 @@ func (kd *KubeDriver) upgrade(chartDir string, valueOpts values.Options, group s
...
+
+	// buf := new(bytes.Buffer)
 	client := action.NewInstall(kd.Configuration)
 	client.Namespace = kd.Namespace()
+	client.RepoURL = chartPathOptions.RepoURL
 	client.DryRun = dry
 	client.ReleaseName = group
 
-	ch, vals, err := kd.getChartAndVals(chartDir, valueOpts)
+	client.ChartPathOptions.CaFile = chartPathOptions.CaFile
```
Comment (sriumcp) [inline driver/kubedriver.go]: Comment out properties we are not using for now.

**Review by sriumcp — CHANGES_REQUESTED** [review]
Please see requested changes.

File: cmd/launch.go (hunk 1 above)
Comment (kalantar) [reply cmd/launch.go]: Set defaults in `action.NewLaunchOpts()` for repoURL and version. Commented out `addChartPathOptionsFlags()`. This removes all CLI options to set the chart path options (including `--repo`).

File: driver/kubedriver.go (hunk 3 above)
Comment (kalantar) [reply driver/kubedriver.go:287]: Left this as is; it is merely copying the values to input for the helm command. These are either explicitly set earlier or are default values.

File: driver/kubedriver.go (hunk 4 above)
Comment (kalantar) [reply driver/kubedriver.go]: Left this as is; it is merely copying the values to input for the helm command. These are either explicitly set earlier or are default values.

File: driver/kubedriver.go (hunk 2 above)
Comment (kalantar) [reply driver/kubedriver.go:290]: see other comments

File: driver/kubedriver.go [hunk 5]
```diff
@@ -323,46 +313,54 @@ func (kd *KubeDriver) upgrade(chartDir string, valueOpts values.Options, group s
...
+	// buf := new(bytes.Buffer)
 	client := action.NewInstall(kd.Configuration)
 	client.Namespace = kd.Namespace()
+	client.RepoURL = chartPathOptions.RepoURL
 	client.DryRun = dry
 	client.ReleaseName = group
 
-	ch, vals, err := kd.getChartAndVals(chartDir, valueOpts)
+	// copy chartPathOptions to client
+	client.ChartPathOptions.CaFile = chartPathOptions.CaFile
```
Comment (sriumcp) [inline driver/kubedriver.go]: Could we please remove all of this? We don't even plan to document how the user is intended to use any of these options ... for now, we are better of without this.

We do need to discuss binary/image/chart signing and verification. We will revisit all of these options at that time.

File: driver/kubedriver.go [hunk 6]
```diff
@@ -300,14 +271,33 @@ func writeManifest(rel *release.Release) error {
...
 // Upgrade a Kubernetes experiment to the next release
-func (kd *KubeDriver) upgrade(chartDir string, valueOpts values.Options, group string, dry bool) error {
+func (kd *KubeDriver) upgrade(chartPathOptions action.ChartPathOptions, chartName string, valueOpts values.Options, group string, dry bool) error {
 	client := action.NewUpgrade(kd.Configuration)
 	client.Namespace = kd.Namespace()
 	client.DryRun = dry
 
-	ch, vals, err := kd.getChartAndVals(chartDir, valueOpts)
+	// copy chartpathoptions
+	client.ChartPathOptions.CaFile = chartPathOptions.CaFile
```
Comment (sriumcp) [inline driver/kubedriver.go]: Could we please remove all of this? We don't even plan to document how the user is intended to use any of these options ... for now, we are better of without this.

We do need to discuss binary/image/chart signing and verification. We will revisit all of these options at that time.

**Review by sriumcp — CHANGES_REQUESTED** [review]
Please see change requests.

File: driver/kubedriver.go (hunk 5 above)
Comment (kalantar) [reply driver/kubedriver.go]: This code is initializing a field (the `ChartPathOptions` field) in the Helm object (`client`). This object is used both to find the chart and to call `RunWithContext()` to do the actual work. We do want to initialize the field. However, we may not need to copy every field. I can comment out the fields that we do not explicitly set.

File: driver/kubedriver.go (hunk 6 above)
Comment (kalantar) [reply driver/kubedriver.go]: same response

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**